R/compile.R

```diff
@@ -1,77 +1,49 @@
 #' Compile Sass to CSS
 #'
-#' Compile Sass to CSS using libSass.
+#' Compile Sass to CSS using LibSass.
 #'
 #'
-#' @param file Path to .scss or .sass Sass file. Note that the libSass compiler
-#'   expects .sass files to use the Sass Indented Syntax.
+#' @param input Raw Sass text or a file path to a .scss or .sass Sass file.
+#'   Note that the LibSass compiler expects .sass files to use the Sass Indented Syntax.
 #' @param options Compiler options for Sass. Please specify options using
 #'   \code{\link{sass_options}}.
 #' @param output Specifies path to output file for compiled CSS.
-#' @param text Character vector of Sass code.
-#'
-#' @return If \code{output = NULL}, the function returns a one element character
-#'   vector of the compiled CSS. If the output path is specified, the compiled
-#'   CSS is written to that file and the function has no return value.
-#'
-#' @note \code{file} and \code{text} cannot be compiled at the same time. Please
-#'   pass one or the other.
-#'
-#' @examples
-#' compile_sass(text = "foo { margin: 122px * .3; }")
-#'
+#' @return If \code{output = NULL}, the function returns a string value
+#'   of the compiled CSS. If the output path is specified, the compiled
+#'   CSS is written to that file.
+#' @seealso \url{http://sass-lang.com/guide}
 #' @export
-compile_sass <- function(file = NULL, options = sass_options(), output = NULL, text = NULL) {
-  if (is.null(file) && is.null(text)) {
-    stop("No input detected. Please supply Sass file or text to compile.")
-  }
-
-  file_input <- FALSE
-  text_input <- FALSE
-
-  if (!is.null(file)) {
-    if (!file.exists(file)) {
-      stop("Input file does not exist.")
-    }
-    file_input <- TRUE
-  }
-
-  if (!is.null(text)) {
-    text_input <- TRUE
-  }
-
-  if (text_input && file_input) {
-    stop("Both file and text were supplied. Please supply only a Sass file ",
-         "or text, not both.")
-  }
-
+#' @examples
+#' sass("foo { margin: 122px * .3; }")
+sass <- function(input = NULL, options = sass_options(), output = NULL) {
```

Comment:
I'd love to see this interface (optional of course):

```
sass("foo.scss", options = sass_options(), vars = list( ))


sass(list(
  list(foo = 1, bar = 2),
  "tony_overrides.scss",
  list(primary_color = "red"),
  "$primary-color: 'green';",
  "foo.scss"
))

// Item 1
$foo: 1;
$bar: 2;
// Item 2
@import "/Users/jcheng/R/3.4/x86_64/tony_overrides.scss"
```

Reply:
(In a second PR)

Reply:
added functionality in this PR